packages/block-library/src/button/edit.js

```diff
@@ -257,18 +281,21 @@ function ButtonEdit( props ) {
 						onChange={ ( {
 							url: newURL = '',
 							opensInNewTab: newOpensInNewTab,
+							nofollow: newNoFollow,
 						} ) => {
 							setAttributes( { url: prependHTTP( newURL ) } );
 
-							if ( opensInNewTab !== newOpensInNewTab ) {
-								onToggleOpenInNewTab( newOpensInNewTab );
-							}
+							updateLinkAttributes(
+								newOpensInNewTab,
+								newNoFollow
+							);
```

Comment:
Could we combine the concept of updating the entire link into the `updateLinkAttributes` method? That will bring it into line with other implementations. This will help pave the way for a future refactoring I'm planning.

Reply:
Sure! Unfortunately I still can't figure an optimized way for this https://github.com/WordPress/gutenberg/pull/54110#discussion_r1318554127 yet 😅

Reply:
I think the optimisation is ok as it is now you've explained it.

Reply:
Ah okay, thanks